src/main/java/com/daml/extensions/damlmavenplugin/MojoBase.java

```diff
@@ -4,18 +4,35 @@
  */
 package com.daml.extensions.damlmavenplugin;
 
-import org.apache.maven.plugin.AbstractMojo;
-
 import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.Scanner;
 
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoFailureException;
+
 public abstract class MojoBase extends AbstractMojo {
 
     protected void redirectOutput(InputStream in) {
-        Scanner scanner = new Scanner(in);
-        while (scanner.hasNextLine()) {
-            getLog().info(scanner.nextLine());
+        try (Scanner scanner = new Scanner(in)) {
+            while (scanner.hasNextLine()) {
+                getLog().info(scanner.nextLine());
+            }
+        }
+    }
+
+    protected Path getDamlYamlFile() throws MojoFailureException {
+        Path cwd = Paths.get("").toAbsolutePath();
+        while (cwd != null) {
+            Path damlYaml = cwd.resolve("daml.yaml");
```

Comment:
it does not make much diff but I like to have private static-s for such "magic strings" (if we can say it is one)

Reply:
(this stands for the delimiters and all the string literals)

Reply:
Made some more improvements, but didn't forget to move the string literal to a constant